src/components/timeline/TrackScreenshots.js

```diff
@@ -300,6 +122,132 @@ class HoverPreview extends PureComponent<HoverPreviewProps> {
   }
 }
 
+type Props = {|
+  +threadIndex: ThreadIndex,
+  +windowId: string,
+|};
+
+export function TimelineTrackScreenshots({ threadIndex, windowId }: Props) {
+  // Selectors
+  const selectors = getThreadSelectors(threadIndex);
+  const { start: rangeStart, end: rangeEnd } = useSelector(getCommittedRange);
+  const isMakingPreviewSelection = useSelector(getIsMakingPreviewSelection);
+  const thread = useSelector(selectors.getRangeFilteredThread);
+  const screenshotsById = useSelector(
+    selectors.getRangeFilteredScreenshotsById
+  );
+  const trackHeight = useSelector(getScreenshotTrackHeight);
+  const screenshots = screenshotsById.get(windowId) || EMPTY_SCREENSHOTS_TRACK;
```

Comment:
The selectors here work in the same way, and the component only re-renders if the selector changes. There is a slight behavior change where the mapStateToProps would shallow equality check the generated object. Here the component re-renders if any selector's return value changes. I had to create the `getIsMakingPreviewSelection` in order to retain proper memoization.

Reply:
If this proves necessary, it looks like we mimic the old way by doing something like:
```js
import { shallowEqual, useSelector } from 'react-redux';

const { isMakingPreviewSelection, thread, screenshotsById } = useSelector((state) => ({
  isMakingPreviewSelection: getIsMakingPreviewSelection(state),
  thread: selectors.getRangeFilteredThread(state),
  screenshotsById: selectors.getRangeFilteredScreenshotsById(state),
  ...
}), shallowEqual)
```
see https://react-redux.js.org/api/hooks#equality-comparisons-and-updates

Reply:
> I had to create the getIsMakingPreviewSelection in order to retain proper memoization.

I think you could just have implemented it here:
```js
const isMakingPreviewSelection = useSelector(state => {
  const { previewSelection } = getProfileViewOptions(state);
  return previewSelection.hasSelection && previewSelection.isModifying;
});
```
instead of globally.
(But I don't really mind)